scripts/bin/unpack_dvd

```diff
@@ -93,29 +109,56 @@ def update_yum_repo(repo)
     end
   end
 
-  begin
-    FileUtils.chown_R('root','apache',repo)
-  rescue Exception => e
-    $stderr.puts("Warning: Could not change permissions on #{repo} to 'root:apache'.")
-    $stderr.puts(e)
+  if Process.uid == 0
+    repo_dirs.each  { |dir|
+      begin
+          FileUtils.chown_R('root',group, dir)
+      rescue Exception => e
+        $stderr.puts("Warning: Could not change permissions on #{dir} to root:#{group}.")
+        $stderr.puts(e)
+      end
+
+      begin
+        FileUtils.chmod_R('g+rX',dir)
+      rescue Exception => e
+        $stderr.puts("Warning: Could not change permissions on #{dir} to 'g+rX'.")
+        $stderr.puts(e)
+      end
+      begin
+        FileUtils.chmod('g+s',dir)
+      rescue Exception => e
+        $stderr.puts("Warning: Could  not set group id permission on #{dir} 'g+s'.")
+        $stderr.puts(e)
+      end
+    }
   end
+end
 
-  if Process.uid == 0
-    begin
-      if Gem::Version.new(RUBY_VERSION) < Gem::Version.new('1.9')
-        # Use Puppet to hack around the lack of symbolic modes in Ruby < 1.9
-        system(%{puppet resource file #{repo} recurse=true mode='g+srX' 2>&1 > /dev/null})
-      else
-        FileUtils.chmod_R('g+srX',repo)
+def sort_noarch(dir, arch)
+# dir = directory you are searching
+# arch = the architecture of the machine
+
+# TODO:  May want to check the name first before doing rpm
+#
+  noarchdir = File.expand_path("../noarch", dir )
+  Dir.mkdir(noarchdir, 0750) unless Dir.exists?(noarchdir)
+  Dir.chdir(dir) do
+    Dir.glob('*.rpm').each do |rpmfile|
+      rpmarch = rpmfile.split('.')[-2]
+      unless [ arch, 'noarch'].include?(rpmarch)
+         rpmarch = `rpm -qp #{rpmfile} --queryformat '%{ARCH}' 2> /dev/null`
```

Comment:
I would probably just do this. There is the possibility that the file is named incorrectly and what's actually in the RPM is all that really matters.

It shouldn't slow things down all that much.